src/shared/sol-conffile.c

```diff
@@ -916,10 +942,178 @@ _load_vector_defaults(void)
+    SOL_INT_CHECK(r, >= (int)sizeof(path), -ENOMEM);
+
+    SOL_DBG("Reading alias file '%s' with precedence equals to %lu",
+        path, precedence);
+
+    r = sol_util_load_file_buffer(path, file_contents);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_json_scanner_init(&scanner, file_contents->data, file_contents->used);
+
+    SOL_JSON_SCANNER_ARRAY_LOOP (&scanner, &value, SOL_JSON_TYPE_OBJECT_START,
+        end_reason) {
+        struct sol_alias_ctx *alias_ctx;
+        struct sol_json_token obj_key, obj_value;
+
+        SOL_JSON_SCANNER_OBJECT_LOOP_NEST (&scanner, &value, &obj_key,
+            &obj_value, end_reason) {
+            struct sol_json_scanner array_scanner;
+            struct sol_json_token array_token;
+
+            if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&obj_key, "aliases")) {
+                sol_json_scanner_init_from_token(&array_scanner, &obj_value);
+
+                SOL_JSON_SCANNER_ARRAY_LOOP (&array_scanner, &array_token,
+                    SOL_JSON_TYPE_STRING, end_reason) {
+                    struct sol_buffer buf;
+
+                    r = sol_json_token_get_unescaped_string(&array_token, &buf);
+                    SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+
+                    slice = sol_vector_append(&aliases);
+                    if (!slice) {
+                        r = -ENOMEM;
+                        sol_buffer_fini(&buf);
+                        goto err_exit;
+                    }
+
+                    r = sol_arena_slice_dup(str_arena, slice,
+                        sol_buffer_get_slice(&buf));
+                    sol_buffer_fini(&buf);
+                    SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+                }
+
+                if (end_reason != SOL_JSON_LOOP_REASON_OK) {
+                    SOL_WRN("Error: Invalid Json.");
+                    r = -EINVAL;
+                    goto err_exit;
+                }
+            } else if (SOL_JSON_TOKEN_STR_LITERAL_EQ(&obj_key, "name")) {
+                r = sol_json_token_get_unescaped_string(&obj_value, &node_name);
+                SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+            }
+        }
+
+        if (end_reason != SOL_JSON_LOOP_REASON_OK) {
+            SOL_WRN("Error: Invalid Json.");
+            r = -EINVAL;
+            goto err_exit;
+        }
+
+        r = -EINVAL;
+        SOL_INT_CHECK_GOTO(node_name.used, == 0, err_exit);
+
+        alias_ctx = sol_vector_append(&node_aliases_map);
+        SOL_NULL_CHECK_GOTO(alias_ctx, err_exit);
+
+        alias_ctx->type_name = sol_arena_strdup_slice(str_arena,
+            sol_buffer_get_slice(&node_name));
+        SOL_NULL_CHECK_GOTO(alias_ctx->type_name, err_exit);
+
+        alias_ctx->precedence = precedence;
+        memcpy(&alias_ctx->aliases, &aliases, sizeof(struct sol_vector));
+
+        aliases.data = NULL;
+        aliases.len = 0;
+        node_name.used = 0;
+    }
+
+    if (end_reason != SOL_JSON_LOOP_REASON_OK) {
+        SOL_WRN("Error: Invalid Json.");
```

Comment:
no need for "Error:", you should show the file name so people can guess what json file is wrong.